shopfloor/services/zone_picking.py

```diff
@@ -535,23 +543,43 @@ def _scan_source_package(self, barcode, confirmation=False):
             message = self.msg_store.package_has_no_product_to_take(barcode)
         return response, message
 
-    def _scan_source_product(self, barcode, confirmation=False):
+    def _get_prefill_qty(self, move_line, qty=False):
+        if self.work.menu.no_prefill_qty:
+            return qty
+        return move_line.product_uom_qty
+
+    def _scan_source_product(self, barcode, qty_done, confirmation=False):
         """Search a product and find available lines for it."""
         message = None
         response = None
         search = self._actions_for("search")
         product = search.product_from_scan(barcode)
+        packaging = self.env["product.packaging"].browse()
+        if not product:
+            packaging = search.packaging_from_scan(barcode)
+            product = packaging.product_id
         if not product:
             return response, message
         move_lines = self._find_location_move_lines(product=product)
         if move_lines:
-            response = self._response_for_set_line_destination(first(move_lines))
+            # Since there's no recover feature in this scenario, do not write
+            # qty_done on the backend, but send it to the frontend in the response
+            # instead
+            move_line = first(move_lines)
+            response = self._response_for_set_line_destination(move_line)
+            if packaging:
+                qty = packaging.qty
+            else:
+                qty = 1.0
+            response["data"]["set_line_destination"]["move_line"][
+                "qty_done"
+            ] = qty_done + self._get_prefill_qty(move_line, qty=qty)
         else:
             response = self._list_move_lines(self.zone_location)
             message = self.msg_store.product_not_found()
         return response, message
 
-    def _scan_source_lot(self, barcode, confirmation=False):
+    def _scan_source_lot(self, barcode, qty_done, confirmation=False):
```

Comment:
if someone has overridden these methods you'll break their code. Make it optional in all methods alike.